### PR TITLE
feat: [pipe-21741]: add api integrations for pr panel and merging

### PR DESCRIPTION
**Pull request by cjlee01**
add api integrations for pr panel and merging
image
image

### CODE REVIEW

File: apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    title: string
+    content: string
+    color: string
+    status: string
+  }>({ title: '', content: '', color: '', status: '' })
+
+  const status = useMemo(() => {
+    let _status: ExecutionState | undefined
+    const _count = { ...DEFAULT_COUNTS }
+    // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts:44]: Can you create a ticket under PIPE project and assign it to me with details about exactly what issue we're seeing here mentioning why do we need to add this statement?

File: apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            console.error('Unrecognized PR check status', check)
+            break
+        }
+      }
+
+      if (_count.error) {
+        _status = ExecutionState.ERROR
+        setColor('text-destructive')
+        setBackground('text-destructive')
+        setMessage(`${_count.error}/${total} ${_count.error === 1 ? 'check' : 'checks'} errored out.`)
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts:76]: Can we move this to a "pluralize" function instead? I see this `${_count.error === 1 ? 'check' : 'checks'}1` bit being repeated multiple times.

File: apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+  findChangeReqDecisions,
+  findWaitingDecisions,
+  processReviewDecision
+} from './utils'
+import { useParams } from 'react-router-dom'
+import { PathParams } from '../../RouteDefinitions'
+import { usePullRequestData } from './context/pull-request-data-provider'
+import { isEmpty } from 'lodash-es'
+import { CodeOwnerReqDecision } from '../../types'
+export const diffdata = {
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx:40]: Is this mock data?

File: apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+              reqCodeOwnerLatestApproval={prPanelData?.reqCodeOwnerLatestApproval}
+              codeOwnerPendingEntries={codeOwnerPendingEntries}
+              codeOwnerApprovalEntries={codeOwnerApprovalEntries}
+              latestCodeOwnerApprovalArr={latestCodeOwnerApprovalArr}
+              minApproval={prPanelData?.minApproval}
+              minReqLatestApproval={prPanelData?.minReqLatestApproval}
+              actions={mockPullRequestActions}
+              resolvedCommentArr={prPanelData?.resolvedCommentArr}
+              requiresCommentApproval={prPanelData?.requiresCommentApproval}
+            />
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx:275]: Do you think this component is too big at this point? 😅

With so much prop drilling, it's better to obtain this info from context.

File: apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx (hunk 3 above)
Comment (cjlee01) [reply apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx:40]: It’s mock data cause I was still working on adding the api for diff data

File: packages/playground/src/components/pull-request/sections/pull-request-changes-section.tsx [hunk 5]
```diff
@@ -237,43 +237,48 @@ const PullRequestChangesSection = ({
...
+          {((minApproval ?? 0) > (minReqLatestApproval ?? 0) ||
+            (!isEmpty(approvedEvaluations) && minReqLatestApproval === 0 && minApproval && minApproval > 0) ||
+            ((minApproval ?? 0) > 0 && minReqLatestApproval === undefined)) && (
+            <div className="ml-4">
+              <div className="flex pt-2 border-t mt-3 ml-2 items-center justify-between">
+                {approvedEvaluations && minApproval && minApproval <= approvedEvaluations?.length ? (
+                  <Text className="flex ml-2">
+                    <CheckCircleSolid className="text-success" />
+                    <Text className="pl-2 text-xs">
+                      {`Changes were approved by ${approvedEvaluations?.length} ${approvedEvaluations?.length === 1 ? 'reviewer' : 'reviewers'}`}
```
Comment (vardanbansal-harness) [inline packages/playground/src/components/pull-request/sections/pull-request-changes-section.tsx:249]: Another candidate for `pluralize` function. If you don't want to implement, we can also use this light weight library https://www.npmjs.com/package/pluralize that would do the job for you.

File: apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+
+  useEffect(() => {
+    refetchCodeOwners()
+  }, [pullReqMetadata, pullReqMetadata?.title, pullReqMetadata?.state, pullReqMetadata?.source_sha])
+
+  // If you need to update activities when activityData changes, use useEffect
+  useEffect(() => {
+    setActivities(activityData)
+  }, [activityData])
+  const currentUser = 'Calvin'
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx:127]: Let's remove PII

File: apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts (hunk 1 above)
Comment (cjlee01) [reply apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts:44]: https://harness.atlassian.net/browse/PIPE-22068

File: apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx (hunk 6 above)
Comment (cjlee01) [reply apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx:127]: changed to 'Default'

File: packages/playground/src/components/pull-request/sections/pull-request-changes-section.tsx (hunk 5 above)
Comment (cjlee01) [reply packages/playground/src/components/pull-request/sections/pull-request-changes-section.tsx]: done

File: apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts (hunk 2 above)
Comment (cjlee01) [reply apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts]: done

File: apps/gitness/src/pages/pull-request/utils.ts [hunk 7]
```diff
@@ -9,3 +11,342 @@ export const processReviewDecision = (
...
+      } else {
+        statusCounts.running += 1
+      }
+    } else if (status === ExecutionState.SUCCESS) {
+      if (required) {
+        statusCounts.successReq += 1
+      } else {
+        statusCounts.succeeded += 1
+      }
+    }
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/pull-request/utils.ts:60]: nit: Can convert to `switch` statement

File: packages/playground/src/components/pull-request/pull-request-list.tsx [hunk 8]
```diff
@@ -171,7 +187,7 @@ export function PullRequestList({ pullRequests, LinkComponent }: PageProps) {
       }),
     [headerFilter]
   )
-
+  console.log(filteredData)
```
Comment (vardanbansal-harness) [inline packages/playground/src/components/pull-request/pull-request-list.tsx:190]: nit: remove console

File: apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx (hunk 4 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx:275]: Looks like it's anyways componentized further down, so should be fine.

File: packages/playground/src/components/pull-request/pull-request-list.tsx (hunk 8 above)
Comment (cjlee01) [reply packages/playground/src/components/pull-request/pull-request-list.tsx:190]: remove

File: apps/gitness/src/pages/pull-request/utils.ts (hunk 7 above)
Comment (cjlee01) [reply apps/gitness/src/pages/pull-request/utils.ts:60]: fixed

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**